llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -219,6 +219,8 @@ class TargetTransformInfo {
   /// Get the kind of extension that an instruction represents.
   static PartialReductionExtendKind
   getPartialReductionExtendKind(Instruction *I);
+  static PartialReductionExtendKind
+  getPartialReductionExtendKind(Instruction::CastOps ExtOpcode);
```

Comment:
What about either replacing `getPartialReductionExtendKind(Instruction *I);` with this one, rather than adding a new interface, Or otherwise changing the implementation of `getPartialReductionExtendKind(Instruction *I)` to use `getPartialReductionExtendKind(Instruction::CastOps)` ?